### PR TITLE
NO-JIRA: test/extended/apiserver: test for checking the healthz endpoints for oauth-apiserver

### PR DESCRIPTION
**Pull request by dinhxuanvu**

### CODE REVIEW

File: test/extended/apiserver/health_endpoints.go [hunk 1]
```diff
@@ -115,3 +164,15 @@ func testPath(transport http.RoundTripper, path string, requiredChecks sets.Set[
 	}
 	return nil
 }
+
+func checkFlatform(ctx context.Context, name string, oc *exutil.CLI) error {
```
Comment (p0lyn0mial) [inline test/extended/apiserver/health_endpoints.go:168]: I would change the signature of this function to `isSupportedPlatform(ctx, name, oc) bool`.
For handling errors inside this function I would use `framework.ExpectNoError(err)`.

File: test/extended/apiserver/health_endpoints.go [hunk 2]
```diff
@@ -115,3 +164,15 @@ func testPath(transport http.RoundTripper, path string, requiredChecks sets.Set[
...
+	isMicroShift, err := exutil.IsMicroShiftCluster(oc.AdminKubeClient())
+	framework.ExpectNoError(err)
+	if isMicroShift {
+		return fmt.Errorf("MicroShift does not have this component %s", name)
```
Comment (p0lyn0mial) [inline test/extended/apiserver/health_endpoints.go:172]: and here i would return `false, "MicroShift does not have this component %s", name"

**Comment by dinhxuanvu** [comment]
@p0lyn0mial Updated. PTAL. Thanks.

**Pull request merged**